src/clarity_vm/database/mod.rs

```diff
@@ -146,6 +162,21 @@ impl BurnStateDB for SortitionDBConn<'_> {
         }
     }
 
+    // This is the one the integration test is calling.
+    fn get_burn_header_hash_using_consensus_hash(
+        &self,
+        height: u32,
+        consensus_hash: &ConsensusHash,
+    ) -> Option<BurnchainHeaderHash> {
+        let bt = backtrace::Backtrace::new();
+        warn!("look2 {:?}", bt);
+        let db_handle = SortitionHandleConn::open_reader_consensus(self, consensus_hash).ok()?;
+        match db_handle.get_block_snapshot_by_height(height as u64) {
+            Ok(Some(x)) => Some(x.burn_header_hash),
+            _ => return None,
+        }
+    }
+
```

Comment:
I think this is right approach. You should be more careful about the error handling here. Basically, we want the "burn header hash" to:

* Return `Some(burn_header_hash)` if the given height is `0 <= height < current_burn_block_height`
* Return `None` if the given height is `>= current_burn_block_height`

Anything else should _panic_ (including errors from the SortitionDB: any such error means the database is corrupt, and the node should not be trying to continue processing Stacks transactions).

Reply:
did this